apps/builder/app/builder/features/style-panel/sections/transforms/transform-utils.ts

```diff
@@ -0,0 +1,381 @@
+import {
+  extractRotatePropertiesFromTransform,
+  extractSkewPropertiesFromTransform,
+  parseCssValue,
+} from "@webstudio-is/css-data";
+import {
+  FunctionValue,
+  toValue,
+  type StyleValue,
+  type TupleValue,
+  type TupleValueItem,
+  type UnitValue,
+} from "@webstudio-is/css-engine";
+import type { DeleteProperty, SetProperty } from "../../shared/use-style-data";
+import type { StyleInfo } from "../../shared/style-info";
+import type { TransformPanel } from "./transforms";
+import { useMemo } from "react";
+
+export type TransformFloatingPanelContentProps = {
+  currentStyle: StyleInfo;
+  propertyValue: TupleValue;
+  setProperty: SetProperty;
+};
+
+const defaultTranslate = "0px 0px 0px";
+const defaultScale = "1 1 1";
+const defaultRotate = "rotateX(0deg) rotateY(0deg) rotateZ(0deg)";
+const defaultSkew = "skewX(0deg) skewY(0deg)";
+
+export const isUnitValue = (value: StyleValue): value is UnitValue => {
+  return value?.type === "unit" ? true : false;
+};
+
+export const useHumaneTransformPropertyValues = (props: {
+  currentStyle: StyleInfo;
+  panel: TransformPanel;
+}): { name: string; value: TupleValue } | undefined => {
+  const { currentStyle, panel } = props;
+
+  const properties: { name: string; value: TupleValue } | undefined =
+    useMemo(() => {
+      switch (panel) {
```

Comment:
This should be a separate function exported from this module, instead of custom hook, useMemo call should be where its being used. 

Our goal is to use the least amount of custom hooks and hooks as often as possible inside components where they are used.